### PR TITLE
App launcher CSS update to use flex box.

### PR DESCRIPTION
**Pull request by matthewcarleton**
## Description

I've updated the app launcher css to use flex box. This allows more flexibility for the list view and grid view.

This has been tested and works in all major browsers. There are some visual differences in IE9, IE 10 (windows 8) but it still functions properly.

## Todos
- [x] cross browser test
- [x] Are you sure it works on IE9?
- [x] Is it reponsive?

## Steps to test or reproduce and link to rawgit
https://rawgit.com/matthewcarleton/patternfly/app-launcher-dist/dist/tests/application-launcher-nav.html.

Also, you may want to please any @mentions you'd like to review.

### CODE REVIEW

File: tests/pages/_includes/widgets/navigation/application-launcher.html [hunk 1]
```diff
@@ -236,7 +236,7 @@
           <li class="applauncher-pf-item" role="menuitem">
             <a class="applauncher-pf-link" href="#">
               <i class="applauncher-pf-link-icon pficon pficon-storage-domain" aria-hidden="true"></i>
-              <span class="applauncher-pf-link-title">Recteque</span>
+              <span class="applauncher-pf-link-title">Lorem ipsum dolor sit amet, consectetur adipisicing elit. Est deserunt praesentium aut at expedita in architecto eius ad voluptatum adipisci ullam, quisquam ut rerum atque voluptatem laudantium ex voluptate minus.</span>
```
Comment (andresgalante) [inline tests/pages/_includes/widgets/navigation/application-launcher.html]: @matthewcarleton Let's remove this and make it a single word

**Review by andresgalante — CHANGES_REQUESTED** [review]
The css looks great

**Review by andresgalante — APPROVED** [review]

**Review by cshinn — APPROVED** [review]
LGTM

**Review by cdcabrera — COMMENTED** [review]
Added an odd number of items. @matthewcarleton 
 @andresgalante Are we're expecting the odd number to span the gap? If so I'm good...   

screen shot 2017-07-20 at 10 15 22 am

**Comment by andresgalante** [comment]
@cdcabrera good catch!
@matthewcarleton Can we fix this?

**Comment by matthewcarleton** [comment]
@andres @cdcabrera This is a result of flexbox. I could rebuild so the flexbox is only applied to the children for the grid.

**Comment by andresgalante** [comment]
@matthewcarleton What about using grid with a flexbox fallback? Am I pushing it?

**Comment by matthewcarleton** [comment]
@cdcabrera I've updated with a nice suggestion from @andresgalante thanks!

**Comment by cshinn** [comment]
Looks great @matthewcarleton!

**Comment by beanh66** [comment]
@matthewcarleton Does this work with the Vertical Nav as well or is that a separate PR? Just wanted to make sure since all the examples in rawgit are horizontal nav. Thanks!

**Comment by matthewcarleton** [comment]
@beanh66 that's a great point. I think a separate PR makes more sense for it. Thanks!

**Comment by beanh66** [comment]
@matthewcarleton Okay great thanks! Just wanted to be sure this is available for both Navs 👍

**Review by beanh66 — APPROVED** [review]
LGTM!

**Review by cdcabrera — APPROVED** [review]
@matthewcarleton looks good

@andresgalante @cshinn

**Pull request merged**